src/indexable.cr

```diff
@@ -412,24 +412,80 @@ module Indexable(T)
     indexes.map { |index| self[index] }
   end
 
+  # Calls the given block once for each index in `self`, passing `self`
+  # and `other` index element as parameter.
+  #
+  # ```
+  # a = [1, 2, 3]
+  # b = ["a", "b", "c"]
+  #
+  # a.zip(b)  { |x,y| puts "[ #{x} -- #{y} ]" }
+  # ```
+  #
+  # produces:
+  #
+  # ```text
+  # [ 1 -- a ]
+  # [ 2 -- b ]
+  # [ 3 -- c ]
+  #```
   def zip(other : Indexable)
     each_with_index do |elem, i|
       yield elem, other[i]
     end
   end
 
+  # Returns an `Array` of Tuples populated with the *N* index
+  # element from `self` and *N* index element from `other`.
+  #
+  # Raises IndexError if `other` has no index corresponding to `self`.
```

Comment:
I'd mention that this happens when `other` is shorter than `self`.

Reply:
Makes sense.  It makes the sentence more clear.